File: grand-vista-backend/src/main/java/com/grandvista/hotel/service/RoomService.java
package com.grandvista.hotel.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.grandvista.hotel.entity.Room;
import com.grandvista.hotel.repository.RoomRepository;

@Service
public class RoomService {

    @Autowired
    private RoomRepository roomRepository;

    public Room addRoom(Room room) {
        return roomRepository.save(room);
    }

    public List<Room> getAllRooms() {
        return roomRepository.findAll();
    }
    
    public List<Room> getAvailableRooms() {

        return roomRepository
                .findByRoomStatus("AVAILABLE");
    }
}

File: grand-vista-frontend/src/services/roomService.js
import api from "./api";

export const getAllRooms = () =>
    api.get("/rooms");

export const addRoom = (data) =>
    api.post("/rooms", data);

export const getAvailableRooms = () =>
    api.get("/rooms/available");

export const getOccupiedRooms = () =>
    api.get("/rooms/occupied");